### PR TITLE
Add updated_time field to project model

### PR DESCRIPTION
**Pull request by dragansah**
https://trello.com/c/dDlUCAgD/599-redo-project-metadata-table
  - updated fields on projects home view
  - updated fields on project open view

### CODE REVIEW

**Comment by dragansah** [comment]
https://trello.com/c/dDlUCAgD/599-redo-project-metadata-table
and
https://trello.com/c/pFlAnnIv/592-replace-data-uploaded-on-with-project-last-updated

File: server/db/versions/010_add_updated_time_field_postgresql_downgrade.sql [hunk 1]
```diff
@@ -0,0 +1 @@
+ALTER TABLE projects DROP COLUMN updated_time;
```
Comment (Ashalynd) [inline server/db/versions/010_add_updated_time_field_postgresql_downgrade.sql]: This is not very nice: that means you'll remove the updated time for all existing projects. Also, I don't get why this change is applied at all, what was the reason for that?

File: server/src/optima/dbmodels.py [hunk 2]
```diff
@@ -46,7 +46,9 @@ class ProjectDb(db.Model):
                                 uselist=False)
     project_data = db.relationship('ProjectDataDb', backref='projects',
                                 uselist=False)
-    creation_time = db.Column(db.DateTime(timezone=True), server_default=text('now()'))
+    creation_time = db.Column(db.DateTime(timezone=True), default=db.func.now())
```
Comment (Ashalynd) [inline server/src/optima/dbmodels.py]: We had updated_time = creation_time initially, which was assumed to be confusing.

**Comment by Ashalynd** [comment]
I suggest not to change the existing structure of the tables, because it does not seem to be needed for this, and can introduce problems for the existing projects.

File: server/db/versions/010_add_updated_time_field_postgresql_downgrade.sql (hunk 1 above)
Comment (Ashalynd) [inline server/db/versions/010_add_updated_time_field_postgresql_downgrade.sql]: Oops, didn't realize that was the new column.

File: server/src/optima/dbmodels.py (hunk 2 above)
Comment (Ashalynd) [inline server/src/optima/dbmodels.py]: This change for creation_time does not make sense, because already existing column in DB is not modified anyway. For updated_time, we don't need the default (because when the project is created, it has not been updated yet).

Comment (tdelev) [inline server/src/optima/dbmodels.py]: The change for `creation_time` it's for consistency reasons, the effect of both is same and I think the later it's more robust because allows defaults on INSERT and UPDATE. Maybe only we could add to the migration script drop defaults
```sql
ALTER COLUMN creation_time DROP DEFAULT;
```
The defaults for `updated_time` it's not problematic, we could remove, but still it's no source of confusion since same `creation_time` and `updated_time` means no update.

Comment (Ashalynd) [inline server/src/optima/dbmodels.py]: Let's not change the existing stuff that is proven to work, also I'd rather not drop DEFAULT value for data creation column just for the sake of ORM :-}

Comment (dragansah) [inline server/src/optima/dbmodels.py]: The change on `creation_time` is for consistency reasons, and also I think using `default` instead of `server_default` is more robust because it allows defaults on INSERT and UPDATE. Only think we could add is one line in the migration script, to align the database:
```sql
ALTER COLUMN creation_time DROP DEFAULT;
```
For `updated_time` the default is not problematic, and can be removed, but still it's not source of confusion because same `creation_time` and `updated_time` means no update.

Comment (dragansah) [inline server/src/optima/dbmodels.py]: The change of the database is required to complete the task, it is the task. And it will not break any existing stuff.

Comment (Ashalynd) [inline server/src/optima/dbmodels.py]: I am ok with introducing new column, but there are no real reasons to change the creation_time (creation_time is only set once for a new record and should never be updated), so I suggest leaving that column the way it was.

Apart from that I'd like to get confirmation from Cliff that he really wants to log every update to the _project_  vs every update to the _model_ :-}

Comment (cliffckerr) [inline server/src/optima/dbmodels.py]: I don't know what distinction you're making between the "project" and the "model"...any time anything is saved the "Last modified" date should be updated.

Comment (Ashalynd) [inline server/src/optima/dbmodels.py]: Project is updated when you e.g. upload a new json.

Model is updated when you run a new calculation and save the result.

Comment (Ashalynd) [inline server/src/optima/dbmodels.py]: Regarding the difference between default and server_default (which we consistently have in many places already :) )  server_default means you don't provide the value to the column and DB fills the default there; default means you specify which default value to provide in 'INSERT' command. I prefer the first approach because it guarantees that there will always be a value if something is inserted.

Comment (cliffckerr) [inline server/src/optima/dbmodels.py]: If you save the result then you modify the json :) I still don't think there's a difference, and "Last updated" should reflect any change in either the "model" or the "project".

**Comment by dragansah** [comment]
@Ashalynd implemented your remarks by returning `server_default` on `creation_date` field and removed `default` on `updated_time`, @cliffckerr here is a screenshot of projects list 
no_mod

**Comment by Ashalynd** [comment]
Thanks!
